rusty-machine/src/learning/gmm.rs

```diff
@@ -132,6 +132,36 @@ impl GaussianMixtureModel {
         }
     }
 
+    /// Constructs a new GMM with the specified prior mixture weights.
+    ///
+    /// The mixture weights must have the same length as the number of components.
+    /// The sum of the mixture weights must be 1.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rusty_machine::learning::gmm::GaussianMixtureModel;
+    /// use rusty_machine::linalg::vector::Vector;
+    ///
+    /// let mix_weights = Vector::new(vec![0.25, 0.25, 0.5]);
+    ///
+    /// let _ = GaussianMixtureModel::with_weights(3, mix_weights);
+    /// ```
+    pub fn with_weights(k: usize, mixture_weights: Vector<f64>) -> GaussianMixtureModel {
+        assert!(mixture_weights.size() == k, "Mixture weights must have length k.");
```

Comment:
>  I don't see a situation where a Result would be more helpful than panicking?

If someone is writing a script that calls this function with a fixed argument, then there is no such situation. But as a library author, you don't know what your callers might want: the guts of some application five years from now could invoke this with `mixture_weights` computed from the result of some other function applied to some other function applied to user input. The terrifying discipline of monadic error handling is precisely that the programmer is forced to explicitly acknowledge through the type system that the entire thread could blow up _right here_ if they're not sure that the argument is correct and they don't supply something else to do instead. It's definitely _not_ the most ergonomic on average. But data-mungers who care about ergonomics more than correctness should really be working in Python.